llvm/lib/Transforms/Yk/ControlPoint.cpp

```diff
@@ -121,8 +122,24 @@ class YkControlPoint : public ModulePass {
     // Get function containing the control point.
     Function *Caller = OldCtrlPointCall->getFunction();
 
-    // Find all live variables just before the call to the control point.
+    // Check that the control point was called inside, what appears to be, an
```

Comment:
This comment is a bit confusing: can we not just say "Check that the control point is inside a loop." Also, does the code below actually check only top-level loops? Do we even need that as a constraint? I guess it doesn't *hurt* to have it as a constraint though.

Reply:
> can we not just say "Check that the control point is inside a loop." 

Not without a more complex check, as we'd have to recursively check sub-loops (LLVM's `LoopInfo`is a tree of loops).

I considered searching deeper, but ended up concluding that it's not really worth the hassle for now.

What do you think?

Reply:
Ah, so this really does check only top-level loops? Fine! Then let's make the comment "Check that the control point is in a top-level loop." without the extra text?

Reply:
Yes please!

Reply:
It appears that the iterator over the loops is already recursing into sub-loops, so if you are OK with it, we can just say "yk_control_point() must be called inside a loop".

What do you think?

Reply:
bd256da implements that.

Reply:
Oh, wow, I've made you made it worse -- sorry! I think we should go back with what you had. I'm a bit stunned there's not a nicer way of doing it!

Reply:
Just to check, are we talking about the `any_of` here, and not whether we search inner loops or not?

The `any_of` is verbose, but actually fairly idiomatic of C++. A seasoned C++ programmer would still prefer what you proposed, I think.

Reply:
Ah, OK, then let's keep it then!